traverson4j-core/src/main/java/uk/co/autotrader/traverson/http/Request.java

```diff
@@ -8,11 +8,11 @@
     private String url;
     private Method method;
     private String acceptMimeType;
-    private Map<String, String> headers;
-    private Map<String, List<String>> queryParameters;
-    private Map<String, List<String>> templateParams;
+    private final Map<String, String> headers;
+    private final Map<String, List<String>> queryParameters;
+    private final Map<String, List<String>> templateParams;
     private Body body;
-    private List<AuthCredential> authCredentials;
+    private final List<AuthCredential> authCredentials;
```

Comment:
Whats your thoughts on these changes? Should we aim to make them all final?

Reply:
Think it would mean we'd have to change a lot more of our code

Reply:
I guess with your changes that we should go in the direction of final fields, to be consistent with the rest of the class, we should probably add a TODO comment to help future developers know what direction to head in.